be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidatorSuite.scala

```diff
@@ -0,0 +1,146 @@
+package ch.epfl.pop.pubsub.graph.validators
+
+import akka.actor.{Actor,ActorSystem,Props}
+import akka.actor.typed.ActorRef
+import akka.pattern.AskableActorRef
+import akka.testkit.{ImplicitSender,TestKit,TestProbe}
+import akka.util.Timeout
+
+import ch.epfl.pop.model.network.{JsonRpcRequest, MethodType}
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.model.network.method.ParamsWithMessage
+import ch.epfl.pop.model.objects.{Base64Data, Channel, LaoData, PrivateKey, PublicKey}
+import ch.epfl.pop.pubsub.graph.{DbActor, ErrorCodes, GraphMessage, PipelineError}
+
+import util.examples.MessageExample
+
+import org.scalatest.{BeforeAndAfterAll,FunSuiteLike,Matchers}
+
+import scala.concurrent.duration.FiniteDuration
+
+import com.google.crypto.tink.subtle.Ed25519Sign
+
+class MessageValidatorSuite extends TestKit(ActorSystem("myTestActorSystem")) with FunSuiteLike with ImplicitSender with Matchers with BeforeAndAfterAll {
+
+    // Implicites for system actors
+    implicit val duration = FiniteDuration(5 ,"seconds")
+    implicit val timeout = Timeout(duration)
+    override def afterAll(): Unit = {
+        // Stops the testKit
+        TestKit.shutdownActorSystem(system)
+    }
+
+    private final val KEYPAIR: Ed25519Sign.KeyPair = Ed25519Sign.KeyPair.newKeyPair
```

Comment:
Same remark I made in `MessageExample`: 
> I would suggest sticking with the same KeyPair instead of generating a new one each time the tests are executed. That > would guarantee that the test data doesn't change for each run and make any existing/future bugs reproducable for
>future test attempts"

Reply:
yeah sounds good, I generated a pair that will now be used as constant Base64Data in both classes